docs/core/whats-new/dotnet-10/libraries.md

```diff
@@ -328,3 +328,363 @@ A community contribution improved the performance of <xref:System.IO.Compression
+
+            using Process process = Process.Start(psi)!;
+            Thread.Sleep(5_000);
+
+            GenerateConsoleCtrlEvent(CTRL_C_EVENT, (uint)process.Id);
+            process.WaitForExit();
+
+            Console.WriteLine("Child process terminated gracefully, continue with the parent process logic if needed.");
+        }
+        else
+        {
+            // If you need to send a CTRL+C, the child process needs to re-enable CTRL+C handling, if you own the code, you can call SetConsoleCtrlHandler(NULL, FALSE).
+            // see https://learn.microsoft.com/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw#remarks
+            SetConsoleCtrlHandler((IntPtr)null, false);
+
+            Console.WriteLine("Greetings from the child process!  I need to be gracefully terminated, send me a signal!");
+
+            bool stop = false;
+
+            var registration = PosixSignalRegistration.Create(PosixSignal.SIGINT, ctx =>
+            {
+                stop = true;
+                ctx.Cancel = true;
+                Console.WriteLine("Received CTRL+C, stopping...");
+            });
+
+            StreamWriter sw = File.AppendText("log.txt");
+            int i = 0;
+            while (!stop)
+            {
+                Thread.Sleep(1000);
+                sw.WriteLine($"{++i}");
+                Console.WriteLine($"Logging {i}...");
+            }
+
+            // Clean up
+            sw.Dispose();
+            registration.Dispose();
+
+            Console.WriteLine("Thanks for not killing me!");
+        }
+    }
+
+    private const int CTRL_C_EVENT = 0;
+    private const int CTRL_BREAK_EVENT = 1;
+
+    [DllImport("kernel32.dll", SetLastError = true)]
+    [return: MarshalAs(UnmanagedType.Bool)]
+    private static extern bool SetConsoleCtrlHandler(IntPtr handler, [MarshalAs(UnmanagedType.Bool)] bool Add);
+
+    [DllImport("kernel32.dll", SetLastError = true)]
+    [return: MarshalAs(UnmanagedType.Bool)]
+    private static extern bool GenerateConsoleCtrlEvent(uint dwCtrlEvent, uint dwProcessGroupId);
+}
+```
+
+### AES KeyWrap with Padding (IETF RFC 5649)
+
+AES-KWP is an algorithm that is occasionally used in constructions like Cryptographic Message Syntax (CMS) EnvelopedData, where content is encrypted once, but the decryption key needs to be distributed to multiple parties, each one in a distinct secret form.
+
+.NET now supports the AES-KWP algorithm via instance methods on the <xref:System.Security.Cryptography.Aes> class:
+
+```csharp
+private static byte[] DecryptContent(ReadOnlySpan<byte> kek, ReadOnlySpan<byte> encryptedKey, ReadOnlySpan<byte> ciphertext)
+{
+    using (Aes aes = Aes.Create())
+    {
+        aes.SetKey(kek);
+
+        Span<byte> dek = stackalloc byte[256 / 8];
+        int length = aes.DecryptKeyWrapPadded(encryptedKey, dek);
+
+        aes.SetKey(dek.Slice(0, length));
+        return aes.DecryptCbc(ciphertext);
+    }
+}
+```
+
+### Post-quantum cryptography updates
```

Comment:
Incorporate all this into the post-quantum cryptography section earlier in this article.